### PR TITLE
Rianfowler/sc 59398/remove withRouter from UploadLicenseFile

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- removes `withRouter` from `UploadLicenseFile` and replaces with hooks from react-router
- `withRouter` is deprecated in react-router 6.0

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/59398/update-kots-web-react-router-5-0-5-1

#### Does this PR introduce a user-facing change?
none-refactor 

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/components/UploadLicenseFile.tsx [hunk 1]
```diff
@@ -0,0 +1,683 @@
+import React, { useEffect, useReducer } from "react";
+import { useHistory } from "react-router";
+import { Link } from "react-router-dom";
+import { KotsPageTitle } from "@components/Head";
+// TODO: upgrade this dependency
+// @ts-ignore
+import Dropzone from "react-dropzone";
```
Comment (rianfowler) [inline web/src/components/UploadLicenseFile.tsx:7]: I tried adding the `@types` but I started getting a bunch of errors. I think we might be using this component the wrong way.

File: web/src/components/UploadLicenseFile.tsx [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+                <CodeSnippet
+                  language="bash"
+                  canCopy={true}
+                  onCopyText={
+                    <span className="u-textColor--success">
+                      Command has been copied to your clipboard
+                    </span>
+                  }
+                >
+                  {typeof licenseExistErrData === "string"
```
Comment (rianfowler) [inline web/src/components/UploadLicenseFile.tsx:663]: added a type guard here. we will need to make sure we refactor this kind of thing out. shouldn't be assigning strings and objects to the same variable.

File: web/src/components/apps/AppLicense.tsx [hunk 3]
```diff
@@ -157,15 +157,19 @@ const AppLicenseComponent = (props: Props) => {
     const airgapLicense = await yaml.safeLoad(contentStr);
     const { appLicense } = state;
 
-    if (airgapLicense.spec?.licenseID !== appLicense?.id) {
+    // TODO: FIX THIS
+    // @ts-ignore
+    if (airgapLicense?.spec?.licenseID !== appLicense?.id) {
```
Comment (rianfowler) [inline web/src/components/apps/AppLicense.tsx:162]: dunno how this got in. going to standardize the makefiles / linting commands next week so hopefully that will block these kinds of things during PR checks

Comment (miaawong) [reply web/src/components/apps/AppLicense.tsx:162]: what is the error?

File: web/src/components/UploadLicenseFile.tsx (hunk 1 above)
Comment (miaawong) [reply web/src/components/UploadLicenseFile.tsx:7]: are you referring to the `js-yaml` package?

**Review by miaawong — APPROVED** [review]

File: web/src/components/apps/AppLicense.tsx (hunk 3 above)
Comment (rianfowler) [reply web/src/components/apps/AppLicense.tsx:162]: the type returned by `TextDecoder` is not the type that `yaml.safeLoad` expects. I'm not sure what any of this is so I'm avoiding changing it for now. 

```
 const content: any = await getFileContent(files[0]);
    const contentStr = new TextDecoder("utf-8").decode(content);
    const airgapLicense = await yaml.safeLoad(contentStr);
    const { appLicense } = state;

    if (airgapLicense.spec?.licenseID !== appLicense?.id) {
    // TODO: FIX THIS
    // @ts-ignore
    if (airgapLicense?.spec?.licenseID !== appLicense?.id) {
```

File: web/src/components/UploadLicenseFile.tsx (hunk 1 above)
Comment (rianfowler) [reply web/src/components/UploadLicenseFile.tsx:7]: I was able to add the types for that one and get it to work but `Dropzone` is like ten versions behind.

**Pull request merged**